src/texttransformation.js

```diff
@@ -0,0 +1,309 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module typing/texttransformation
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import TextWatcher from '@ckeditor/ckeditor5-utils/src/textwatcher';
```

Comment:
I think that TextWatcher, since it knows about the editor, would be better here in `ckeditor5-typing`. Semantically, it's a typing util.